lib/custom_image_crop_widget.dart

```diff
@@ -0,0 +1,255 @@
+            ),
+            if (widget.drawPath != null) ...{
+              widget.drawPath(path),
+            },
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onScaleStart(_) {
+    dataTransitionStart = null; // Reset for update
+  }
+
+  void onScaleUpdate(ScaleEvent event, Path path) {
+    if (dataTransitionStart != null) {
+      addTransition(dataTransitionStart - CropImageData(scale: event.scale, angle: event.rotationAngle));
+    }
+    dataTransitionStart = CropImageData(scale: event.scale, angle: event.rotationAngle);
+  }
+
+  void onMoveStart(_) {
+    dataTransitionStart = null; // Reset for update
+  }
+
+  void onMoveUpdate(MoveEvent event, Path path) {
+    addTransition(CropImageData(x: event.delta.dx, y: event.delta.dy));
+  }
+
+  Path getPath(double cropWidth, double width, double height) {
+    if (widget.path != null) {
+      return widget.path(cropWidth);
+    }
+    switch (widget.shape) {
+      case CustomCropShape.Circle:
+        return Path()
+          ..addOval(
+            Rect.fromCircle(
+              center: Offset(width / 2, height / 2),
+              radius: cropWidth / 2,
+            ),
+          );
+      default:
+        return Path()
+          ..addRect(
+            Rect.fromCenter(
+              center: Offset(width / 2, height / 2),
+              width: cropWidth,
+              height: cropWidth,
+            ),
+          );
+    }
+  }
+
+  @override
+  Future<MemoryImage> onCropImage() async {
+    if (imageAsUIImage == null) {
+      return null;
+    }
+    final cropWidth = min(width, height) * widget.cropPercentage;
+    final pictureRecorder = ui.PictureRecorder();
+    final canvas = Canvas(pictureRecorder);
+    final defaultScale = min(imageAsUIImage.width / cropWidth, imageAsUIImage.height / cropWidth);
+    final scale = data.scale * defaultScale / 2;
+    final clipPath = Path.from(getPath(cropWidth, cropWidth, cropWidth));
+    final matrix4Image = Matrix4.diagonal3(vector_math.Vector3(1, 1, 0))
+      ..translate(data.x + cropWidth / 2, data.y + cropWidth / 2)
+      ..scale(scale)
+      ..rotateZ(data.angle);
+    final bgPaint = Paint()
+      ..color = widget.backgroundColor
+      ..style = PaintingStyle.fill;
+    canvas.drawRect(Rect.fromLTWH(0, 0, cropWidth, cropWidth), bgPaint);
+    canvas.save();
+    canvas.clipPath(clipPath);
+    final paint = Paint()..isAntiAlias = false;
+    canvas.transform(matrix4Image.storage);
+    canvas.drawImage(imageAsUIImage, Offset(-imageAsUIImage.width / 2, -imageAsUIImage.height / 2), paint);
+    canvas.restore();
+    ui.Picture picture = pictureRecorder.endRecording();
+    ui.Image image = await picture.toImage(cropWidth.floor(), cropWidth.floor());
+
+    // Optionally remove magenta from image by evaluating every pixel
+    // See https://github.com/brendan-duncan/image/blob/master/lib/src/transform/copy_crop.dart
+
+    ByteData bytes = await image.toByteData(format: ui.ImageByteFormat.png);
```

Comment:
run in isolate

Reply:
unclear if this already runs in an isolate. Could be impossible since it cannot be shared across isolates

Reply:
Didn't work, so for now we will not implement this. Workaround would include saving and loading the image...